### ISSUE TITLE
Come Up With Experiments For The Report

### ISSUE THREAD

**Issue by andydmh** [andydmh/Machine-Learning-Group9#15]
Some ideas for experiments that we may include in the report. For example what type of comparisons with other models we will do (example: which one makes better prediction based on the loss), visualizations that we may want to include, analyzes of the training behavior of our model.

**Comment by andydmh** [comment]
@jbowren, @Jagruteep, @vperidot we are currently training our model with 3 datasets (check the dataset section for more info) 2 of them are small and easy to load, the other one is the Ames Housing Dataset. The performance of our current model trained with Batch Gradient Descent is not great on Ames Housing Dataset, this may be because either we may need to extract more features or because of the complexity of the dataset. We are already comparing with one scikit-learn library (Ridge).

**Comment by andydmh** [comment]
@Jagruteep, @jbowren, @vperidot take a look at run.py there are some experiments in there.

**Comment by Jagruteep** [comment]
Ames Housing Dataset may require a Stochastic Gradient Descent because of the large training set.

**Comment by jbowren** [comment]
Andy spoke with Dr. Hollander, who said it is fine to just use Batch Gradient Descent, though we can continue onto Stochastic Gradient Descent if time permits.

**Comment by Jagruteep** [comment]
Do we have to include the implementation on all three datasets in a report?

**Comment by jbowren** [comment]
We can include as many as we want. We have three in total, but for the moment I am only including two in the report because I was having a hard time finding a citation. There is the webpage listed in the description file of the data folder, but it is not an actual study.

It's an easy addition. If you find a source to cite, go ahead and put the citation in the references.

**Comment by andydmh** [comment]
@Jagruteep, @jbowren, @vperidot everything is ready in terms of comparing against skit-learn libraries (we are comparing against SGD and a method called Ridge Regression). Also I updated the results on the Bostom House Pricing dataset and the Diabetes dataset. If you check inside the results directory you will find a file called report.txt in which you will find all the information about the experiment. In there you will also find a training_behavior.png which shows how our Batch Gradient Descent method optimizes (minimizes) the Mean Squared Error on the training and validation set. We are only missing  training with the Ames Housing dataset, when we load it now it only contains 0s (try to double check that)

**Comment by jbowren** [comment]
I found an issue in the update rule, the second term inside the for loop of dJ should have been on the outside. I fixed it and got results comparable with the SKLearn ridge regression model on the diabetes dataset: 2642.49 vs 2443.06. It also beat SKLearn's SGDRegressor, which had an MSE of 3084.57. It also was comparable on the boston housing dataset and once again beat SGDRegressor.

I'm now applying the model to the Ames Housing dataset, but incorporating all of the data is proving difficult. With the first 5 numeric values, the loss went down at a snails pace, partly because I had to decrease the learning rate by a ton. With the first 20 numeric values, it goes down quicker, but still slowly. After 2000 epochs, the loss is still huge for both our implementation and that of SKLearn:

The mean squared error of the implemented model is: 2087115933.34
The mean squared error of the Ridge sklearn model is: 1434486064.96

How important is it that we include this dataset in our analysis? Maybe just including the other two would be fine?

**Comment by andydmh** [comment]
@jbowren yes I think that including the once in which we are getting reasonable results is fine. The Boston House dataset is very similar to the other one so it should be fine to include the Diabetes and Boston House datasets on the report.
